### ISSUE TITLE
ci: verify Helm Chart with snapshot tests

### ISSUE THREAD

**Issue by apricote** [hetznercloud/hcloud-cloud-controller-manager#688]
### TL;DR

Verify the functionality of the Helm Chart by adding snapshots for some example value files. This showcases changes to the rendered manifests.

### Expected behavior

Same as in `csi-driver`:

- Committed snapshots: https://github.com/hetznercloud/csi-driver/tree/main/chart/.snapshots
- Script to update snapshots: https://github.com/hetznercloud/csi-driver/blob/main/hack/update-helm-snapshots.sh
- Tests for up-to-date snapshots in CI: https://github.com/hetznercloud/csi-driver/blob/4c77198f9138dfc672cedea184551de2c20857b0/.github/workflows/test.yml#L132-L155

We can start with the values.yaml we have right now and possibly add one similar to the example-prod in csi-driver.
